src/python/pants/backend/go/util_rules/analyze_package.go

```diff
@@ -207,11 +202,16 @@ func analyzePackage(directory string, buildContext *build.Context) (*Package, er
 		if err != nil {
 			pkg.InvalidGoFiles[name] = err.Error()
 			// Fall-through to allow still listing the file's existence.
+			// TODO: This should just be I/O errors now, so consider erroring out here as that is infra failure.
```

Comment:
Yeah, not sure if that should blow up Pants or not. I lean towards yes in case it's a bug we should be aware of?

Reply:
Probably because it means the source file was not available in the sandbox which is not an issue with source code but a bug with Pants.